File: 240402/국영수 순이지/korean-english-math-order.java
import java.util.*;

class Student implements Comparable<Student> {
    String name;
    int kor, eng, math;

    Student (String name, int kor, int eng, int math) {
        this.name = name;
        this.kor = kor;
        this.eng = eng;
        this.math = math;
    }

    public void printInfo() {
        System.out.println(this.name + " " + this.kor + " " + this.eng + " " + this.math);
    }

    @Override
    public int compareTo(Student student) {
        if(this.kor == student.kor) {
            if(this.eng == student.eng) 
                return student.math - this.math;
            return student.eng - this.eng;
        }
        return student.kor - this.kor;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Student[] s = new Student[n];

        for(int i = 0; i < n; i ++) {
            s[i] = new Student(sc.next(), sc.nextInt(), sc.nextInt(), sc.nextInt());
        }

        Arrays.sort(s);

        for(int i = 0; i < n; i ++) {
            s[i].printInfo();
        }
    }
}

File: 240614/국영수 순이지/korean-english-math-order.py
class Student:
    def __init__(self, name, kor, eng, math):
        self.name = name
        self.kor = kor
        self.eng = eng
        self.math = math
    
    def __repr__(self):
        return f'{self.name} {self.kor} {self.eng} {self.math}'

n = int(input())
students = []

for i in range(n):
    s = input().split()
    students.append(Student(s[0], int(s[1]), int(s[2]), int(s[3])))

students.sort(lambda x: (-x.kor, -x.eng, -x.math))

for i in range(n):
    print(students[i])